src/lib/fp.ts

```diff
@@ -4,4 +4,22 @@ import * as Option from 'fp-ts/Option';
 export const pipe = F.pipe;
 export const combineOptions = Option.sequenceArray;
 
-export const nullableMap = <T, U>(arg: T | null, f: (x: T) => U) => (arg === null ? null : f(arg));
+export function nullableMap<T, U>(arg: T | null, f: (x: T) => U): U | null;
+export function nullableMap<T, U>(f: (x: T) => U): (arg: T | null) => U | null;
+export function nullableMap<T, U>(funOrValue: (T | null) | ((x: T) => U), fun?: (x: T) => U) {
+    function helper(arg: T | null, f: (x: T) => U) {
+        return arg === null ? null : f(arg);
+    }
+    function isFunction(f: (T | null) | ((x: T) => U)): f is (x: T) => U {
+        return typeof f === 'function';
+    }
+
+    if (typeof fun !== 'undefined' && typeof funOrValue !== 'function') {
+        return helper(funOrValue, fun);
+    }
+    if (isFunction(funOrValue)) {
+        return (arg: T | null) => helper(arg, funOrValue);
+    }
+    // Dette skal aldri skje, men jeg fant ikke noen måte å få inn det i typingen på.
```

Comment:
Kan det fungere hvis vi splitter opp i 2 funksjoner der man ikke har optionals?

Reply:
Tenker du å splitte `nullableMap` i to (f.eks. `nullableMap` og `nullableMapCurried`)? I så fall så vil det funke fint ja, men det jeg ønsket å få til her var å faktisk overloade funksjonen.
Eller misforstod jeg?

Godt mulig det går an å fikse typingen på en måte som gjør at TS skjønner at vi har exhaustive sjekk på det altså, men som kommentaren pekte på så klarte ikke jeg å finne ut hvordan :)

Reply:
Tenkte splitte i 2, hvor 1 overload, med samme navn, men det er kanskje ikke mulig i ts/js.

Reply:
Nei, tror dessverre ikke det :/